Row.h

```diff
@@ -0,0 +1,141 @@
+   bool updated;
+
+   /*
+    * Internal state for tree-mode.
+    */
+   int32_t indent;
+   unsigned int tree_depth;
+} Row;
+
+typedef unsigned int RowField;
+
+typedef Row* (*Row_New)(const struct Machine_*);
+typedef bool (*Row_Filtered)(const Row*, const struct Table_*);
+typedef bool (*Row_Obscured)(const Row*, const struct Table_*);
+typedef int (*Row_CompareByParent)(const Row*, const Row*);
+
+int Row_compare(const void* v1, const void* v2);
+void Row_delete(Object* cast);
+int Row_compare(const void* v1, const void* v2);
+
+typedef struct RowClass_ {
+   const ObjectClass super;
+   const Row_Filtered filtered;
+   const Row_Obscured obscured;
+   const Row_CompareByParent compareByParent;
+} RowClass;
+
+#define As_Row(this_)  ((const RowClass*)((this_)->super.klass))
+
+#define Row_filtered(r_, t_)  (As_Row(r_)->filtered ? (As_Row(r_)->filtered(r_, t_)) : false)
+#define Row_obscured(r_, t_)  (As_Row(r_)->obscured ? (As_Row(r_)->obscured(r_, t_)) : false)
+#define Row_compareByParent(r1_, r2_)  (As_Row(r1_)->compareByParent ? (As_Row(r1_)->compareByParent(r1_, r2_)) : Row_compareByParent_Base(r1_, r2_))
+
+#define ONE_K 1024UL
+#define ONE_M (ONE_K * ONE_K)
+#define ONE_G (ONE_M * ONE_K)
+#define ONE_T (1ULL * ONE_G * ONE_K)
+#define ONE_P (1ULL * ONE_T * ONE_K)
+
+#define ONE_DECIMAL_K 1000UL
+#define ONE_DECIMAL_M (ONE_DECIMAL_K * ONE_DECIMAL_K)
+#define ONE_DECIMAL_G (ONE_DECIMAL_M * ONE_DECIMAL_K)
+#define ONE_DECIMAL_T (1ULL * ONE_DECIMAL_G * ONE_DECIMAL_K)
+#define ONE_DECIMAL_P (1ULL * ONE_DECIMAL_T * ONE_DECIMAL_K)
+
+/* Takes number in bytes (base 1024). Prints 6 columns. */
+void Row_printBytes(RichString* str, unsigned long long number, bool coloring);
+
+/* Takes number in kilo bytes (base 1024). Prints 6 columns. */
+void Row_printKBytes(RichString* str, unsigned long long number, bool coloring);
+
+/* Takes number as count (base 1000). Prints 12 columns. */
+void Row_printCount(RichString* str, unsigned long long number, bool coloring);
+
+/* Takes time in hundredths of a seconds. Prints 9 columns. */
+void Row_printTime(RichString* str, unsigned long long totalHundredths, bool coloring);
+
+/* Takes rate in bare unit (base 1024) per second. Prints 12 columns. */
+void Row_printRate(RichString* str, double rate, bool coloring);
+
+void Row_printLeftAlignedField(RichString* str, int attr, const char* content, unsigned int width);
+
+void Row_printPercentage(float val, char* buffer, int n, uint8_t width, int* attr);
+
+void Row_display(const Object* cast, RichString* out);
+
+extern const RowClass Row_class;
+
+void Row_init(Row* this, const struct Machine_* host);
+
+void Row_toggleTag(Row* this);
+
+static inline int Row_idEqualCompare(const void* v1, const void* v2) {
+   const int p1 = ((const Row*)v1)->id;
+   const int p2 = ((const Row*)v2)->id;
+   return p1 != p2; /* return zero when equal */
+}
+
+static inline int Row_getGroupOrParent(const Row* this) {
```

Comment:
Got some docs for where this is used and what for?

Reply:
Added.